### PR TITLE
Use prettyprinter instead of ansi-wl-pprint

### PR DESCRIPTION
**Pull request by jasagredo**
I didn't do anything fancy, just "inlining" what I found in `prettyprinter-compat-ansi-wl-pprint`.

### CODE REVIEW

File: src/Data/TreeDiff/Class.hs [hunk 1]
```diff
@@ -109,7 +109,7 @@ import Data.These (These (..))
 -- primitive
 import qualified Data.Primitive as Prim
 
-#if MIN_VERSION_base(4,9,0)
+#if MIN_VERSION_base(4,17,0)
```
Comment (jasagredo) [inline src/Data/TreeDiff/Class.hs:112]: I'm unsure if I was doing something weird, but this module seems to only exist in base >=4.17, that's why I updated this cpp

**Review by phadej — COMMENTED** [review]
I won't accept this change as such. It's a breaking change.

I have thought about how to do this and the streps are:
- add `prettyprinter` dependency (behind a manual flag) and functionality similarly how there are printers using `pretty` and `ansi-wl-pprint`.
- make `ansi-wl-pprint` dependency behind a flag too (enabled)
- make `ansi-wl-pprint` flag disabled by default. Later, as that will be a breaking change.

**Comment by 9999years** [comment]
Any way to move forward on this? The rest of the ecosystem has moved on but `tree-diff` has been broken for a year+

**Comment by 9999years** [comment]
@phadej See #91, an upstream change in `ansi-wl-pprint` means that `show` no longer renders colored output correctly:

> The [`ansi-wl-pprint` changelog for 1.0.2](https://github.com/ekmett/ansi-wl-pprint/blob/master/Changelog.md#102) says that "Using `show` won't preserve formatting anymore, as `prettyprinter`s `Show Doc` instance is annotation invariant."

The only way to fix this is by using `Prettyprinter.Render.Terminal.renderStrict` to render output, which is not exposed from `ansi-wl-pprint`. I can't see a way around adding an `prettyprinter-ansi-terminal` dependency.

**Pull request closed without merging**